communication.js

```diff
@@ -159,6 +160,38 @@ function read(device) {
   return response;
 }
 
+/**
+ * Generic helper function that handles resonse messages from the BitBox.
+ */
+function handleResponse(device, version, encryptionKey, authenticationKey, callback) {
+  let response = read(device);
+  console.log(JSON.stringify(response));
+  if (response.ciphertext) {
+    let decodedBytes = Buffer.from(response.ciphertext, 'base64')
+
+    let decrypt = function(decodedBytes) {
+      cryptography.decryptAES(encryptionKey, decodedBytes, function(data) {
+        console.log("=> " + data)
+        callback(JSON.parse(data));
+      });
+    }
+
+    if (version && semver.gte(version, '5.0.0')) {
```

Comment:
`decrypt` isn't executed, it's stored in a variable so that I don't have to write the function twice. I execute it for <5.0.0 firmware BitBoxes in line 187, but for v5.0.0 devices I pass it to the checkHMAC function. It's executed in `cryptography.js:128`

Reply:
I guess I could pull it out into a separate function to make it more readable.

Reply:
disregard, I did not look properly!